build/deploy.sh

```
@@ -0,0 +1,103 @@
+fi
+
+SHA="`git rev-parse HEAD`"
+BRANCH="`git rev-parse --abbrev-ref HEAD`"
+if [ "$BRANCH" == "HEAD" ]; then # Travis does this for some reason
+    BRANCH=$TRAVIS_BRANCH
+fi
+
+if [ "$BRANCH" == "master" -a "$TRAVIS_PULL_REQUEST" != "false" -a "$TRAVIS_PULL_REQUEST" != "" ]; then
+    echo "Skipping deploy for a pull request; the branch build will suffice"
+    exit 0
+fi
+
+BACK_TO_LS_LINK="<a href=\"/\" id=\"commit-snapshot-link\">Go to the living standard</a>"
+SNAPSHOT_LINK="<a href=\"/commit-snapshots/$SHA/\" id=\"commit-snapshot-link\">Snapshot as of this commit</a>"
+
+echo "Branch = $BRANCH"
+echo "Commit = $SHA"
+echo ""
+
+rm -rf $WEB_ROOT || exit 0
+
+if [ $BRANCH != "master" ] ; then
+    # Branch snapshot, if not master
+    BRANCH_DIR=$WEB_ROOT/$BRANCHES_DIR/$BRANCH
+    mkdir -p $BRANCH_DIR
+    curl https://api.csswg.org/bikeshed/ -f -F file=@$INPUT_FILE -F md-status=LS-BRANCH \
+         -F md-warning="Branch $BRANCH $BRANCH_URL_BASE$BRANCH replaced by $LS_URL" \
+         -F md-title="$TITLE (Branch Snapshot $BRANCH)" \
+         -F md-Text-Macro="SNAPSHOT-LINK $SNAPSHOT_LINK" \
+         > $BRANCH_DIR/index.html;
+    echo "Branch snapshot output to $WEB_ROOT/$BRANCHES_DIR/$BRANCH"
+else
+    # Commit snapshot, if master
+    COMMIT_DIR=$WEB_ROOT/$COMMITS_DIR/$SHA
+    mkdir -p $COMMIT_DIR
+    curl https://api.csswg.org/bikeshed/ -f -F file=@$INPUT_FILE -F md-status=LS-COMMIT \
+         -F md-warning="Commit $SHA $COMMIT_URL_BASE$SHA replaced by $LS_URL" \
+         -F md-title="$TITLE (Commit Snapshot $SHA)" \
+         -F md-Text-Macro="SNAPSHOT-LINK $BACK_TO_LS_LINK" \
+         > $COMMIT_DIR/index.html;
+    echo "Commit snapshot output to $WEB_ROOT/$COMMITS_DIR/$SHA"
+    echo ""
+
+    # Living standard, if master
+    curl https://api.csswg.org/bikeshed/ -f -F file=@$INPUT_FILE \
+         -F md-Text-Macro="SNAPSHOT-LINK $SNAPSHOT_LINK" \
+         > $WEB_ROOT/index.html
+    cp .htaccess $WEB_ROOT/.htaccess 2>/dev/null || :
```

This copies `.htaccess` but ignores any errors. This was my best attempt at "copy `.htaccess` if it exists, otherwise don't bother."

I guess maybe instead i should do the equivalent of `if (htaccess exists) { copy it }`. The problem with that is that it's susceptible to race conditions where htaccess disappears or appears between the if and the copy. But that's not realistically a problem in this scenario so I guess it's fine.

If we have a repo with no other resources to test on then that might be better for first pass.